### PR TITLE
#2522 - Concrete projection of interval x zonotope

### LINKED ISSUE #2522: Concrete projection of interval x zonotope
```julia
X = Interval(1, 2) × convert(Zonotope, Hyperrectangle(zeros(2), ones(2)))

LazySets.project(X, [1, 3])

AssertionError: package 'Polyhedra' not loaded (it is required for executing `linear_map with elimination`)

```

but the result is always a zonotope.

### PR DESCRIPTION
**Pull request by SebastianGuadalupe**
Closes #2522.

### CODE REVIEW

File: src/LazyOperations/CartesianProduct.jl [hunk 1]
```diff
@@ -325,3 +325,31 @@ function project(cp::CartesianProduct{N, <:Interval, <:AbstractHyperrectangle{N}
...
+function project(cp::CartesianProduct{N, <:Interval, <:AbstractZonotope{N}}, block::AbstractVector{Int}) where {N}
+    block_vec = collect(block)
+    Z = cp.Y
+    if 1 ∉ block_vec
+        block_vec .-= 1
+        M = projection_matrix(block_vec, dim(Z), N)
+    else
+        M = projection_matrix(block_vec, dim(cp), N)
+        Z = convert(Zonotope, cp)
+    end
```
Comment (schillic) [inline src/LazyOperations/CartesianProduct.jl:353]:
```suggestion
    else
        Z = convert(Zonotope, cp)
    end
    M = projection_matrix(block_vec, dim(Z), N)
```
This is equivalent but saves one line. Your call :smiley:

File: test/unit_CartesianProduct.jl [hunk 2]
```diff
@@ -387,6 +387,10 @@ for N in [Float64, Float32]
     cp = Interval(N(0), N(2)) × Hyperrectangle(N[2, 3], N[1, 1])
     @test project(cp, 1:2) == Hyperrectangle(N[1, 2], N[1, 1])
     @test project(cp, 2:3) == Hyperrectangle(N[2, 3], N[1, 1])
+
+    cp = Interval(N(0), N(2)) × Zonotope(N[2, 3], N[1 0; 0 1])
+    @test isequivalent(project(cp, 1:2), Zonotope(N[1, 2], N[1 0; 0 1]))
+    @test isequivalent(project(cp, 2:3), Zonotope(N[2, 3], N[1 0; 0 1]))
```
Comment (schillic) [inline test/unit_CartesianProduct.jl:393]: Just to understand: is the `isequivalent` needed? From the code I would expect that you have equality here.

**Review by schillic — APPROVED** [review]

File: test/unit_CartesianProduct.jl (hunk 2 above)
Comment (SebastianGuadalupe) [reply test/unit_CartesianProduct.jl:393]: There are cases where the zonotope that the function gives has sparce matrices, so to have a cleaner code I used `isequivalent`

Comment (schillic) [reply test/unit_CartesianProduct.jl:393]: I see, thanks!

**Comment by mforets** [comment]
there is an (unrelated) error in julia v1.0, i'll address it in a separate PR..

**Pull request merged**